### ISSUE TITLE
[Feature request] sync --abort-if / --max-deletes

### ISSUE THREAD

**Issue by flixajki** [ncw/rclone#1032]
--abort-if NUM

--abort-if 0 
If there are no files present in source do not delete files in dest. 

--abort-if 10
If there are less then 10 files in source do not delete files in dest. 

**rsync** 
max-delete=NUM
              This tells rsync not to delete more than NUM files  or  directo‐
              ries.   If  that  limit  is  exceeded, all further deletions are
              skipped through the end of the transfer.  At the end, rsync out‐
              puts  a warning (including a count of the skipped deletions) and
              exits with an error code of 25 (unless some more important error
              condition also occurred).

              Beginning  with version 3.0.0, you may specify --max-delete=0 to
              be warned about any extraneous files in the destination  without
              removing any of them.  Older clients interpreted this as "unlim‐
              ited", so if you don’t know what version the client is, you  can
              use  the  less  obvious --max-delete=-1 as a backward-compatible
              way to specify that no deletions be allowed (though  really  old
              versions didn’t warn when the limit was exceeded).

**Comment by ishuah** [comment]
> --abort-if 0
If there are no files present in source do not delete files in dest.

> --abort-if 10
If there are less then 10 files in source do not delete files in dest.

Are we considering files and directories or just files?
@ajkis

**Comment by wlerin** [comment]
Probably not the right place to post this, but I would really appreciate a `--no-delete` or `--delete-never` option when syncing. Or any other way to suppress the error thrown `--max-deletes 0`.

And before you suggest using copy instead, I need `--track-renames`.

**Comment by xdanger** [comment]
This works for me (rclone v1.42) 
`--max-delete 0 --delete-after`

**Comment by ishuah** [comment]
@ncw @ajkis `--max-delete 0 --delete-after` works for this issue, (as described).
My sample test:
```
$ mkdir a
$ touch a/x.txt
$ touch a/y.txt
$ touch a/z.txt
$ mkdir b
$ tree a
a
├── x.txt
├── y.txt
└── z.txt

0 directories, 3 files
$ tree b

0 directories, 0 files
$ rclone sync --max-delete 0 --delete-after -v b a
2018/08/23 18:22:55 NOTICE: Config file "/Users/ishuahkariuki/.config/rclone/rclone.conf" not found - using defaults
2018/08/23 18:22:55 INFO  : Local file system at /Users/ishuahkariuki/go/src/github.com/ncw/lab/a: Waiting for checks to finish
2018/08/23 18:22:55 INFO  : Local file system at /Users/ishuahkariuki/go/src/github.com/ncw/lab/a: Waiting for transfers to finish
2018/08/23 18:22:55 INFO  : Waiting for deletions to finish
2018/08/23 18:22:55 ERROR : Got fatal error on delete: --max-delete threshold reached
2018/08/23 18:22:55 ERROR : Got fatal error on delete: --max-delete threshold reached
2018/08/23 18:22:55 ERROR : Got fatal error on delete: --max-delete threshold reached
2018/08/23 18:22:55 ERROR : Cancelling sync due to fatal error: failed to delete 3 files
2018/08/23 18:22:55 ERROR : Local file system at /Users/ishuahkariuki/go/src/github.com/ncw/lab/a: not deleting directories as there were IO errors
2018/08/23 18:22:55 ERROR : Fatal error received - not attempting retries
2018/08/23 18:22:55 Failed to sync: failed to delete 3 files
$ tree a
a
├── x.txt
├── y.txt
└── z.txt

0 directories, 3 files
$ tree b
b

0 directories, 0 files 
```
Destination files are not deleted even when source is empty (using `--max-delete 0 --delete-after`).
The `--max-delete` flag was added in ab8c0a81.

**Comment by craig-miskell-fluxfederation** [comment]
Or is https://github.com/ncw/rclone/issues/952 relevant (use copy, not sync)?

**Comment by ishuah** [comment]
@craig-miskell-fluxfederation I see your point and as you noted, #952 (use copy instead of sync) is the best approach when you don't want your source files deleted.

**Comment by wlerin** [comment]
@ishuah @craig-miskell-fluxfederation It's not relevant for my specific use case, though if you don't need to track renamed files then yes, `copy` works fine.

**Comment by olfway** [comment]
> I have a lot more files and data on the cloud than I can store locally, but it is much easier to batch rename (or move) local files and then "sync" those changes with the cloud.

I have a similar use case
Right now I use "sync --max-delete=0" but it ends with an errors, which looks confusing

```
2021-07-12 17:53:12 ERROR : Got fatal error on delete: --max-delete threshold reached
2021-07-12 17:53:12 ERROR : Cancelling sync due to fatal error: failed to delete 1 files
2021-07-12 17:53:12 ERROR : S3 bucket some-bucket: not deleting directories as there were IO errors
2021-07-12 17:53:12 ERROR : Fatal error received - not attempting retries
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Errors:                 2 (fatal error encountered)
Checks:                31 / 31, 100%
Deleted:                1 (files), 0 (dirs)
Elapsed time:        29.9s
2021/07/12 17:53:12 Failed to sync with 2 errors: last error was: failed to delete 1 files
```

I would prefer to add --delete-never to tell rclone just sync local files and do not try to delete

**Comment by Masamune3210** [comment]
correct me if im misunderstanding, but a sync without deletes is just a copy right?

**Comment by wlerin** [comment]
No. As already several times mentioned above, `--track-renames` only works with `sync`.

**Comment by tooptoop4** [comment]
this is important! i have found s3 copy between 2 buckets ends up doing server side copy and ALWAYS re-copies even though files were already same

**Comment by asantoni** [comment]
Just to warn anyone who read what @ishuah and @Masamune3210 mentioned - **rclone `copy` can currently delete files**, so be forewarned: https://github.com/rclone/rclone/issues/6142

**Comment by Masamune3210** [comment]
It only deletes files on the target due to a bug with detecting failed copies in an attempt to avoid leaving partial corrupted copies of files on some backends in certain circumstances, but the warning is still appreciated

**Comment by ncw** [comment]
v1.63 will no longer delete files on copy @asantoni @Masamune3210 - by default it will download to a `.partial` file and rename it if it is correct after the transfer.
